### PR TITLE
Get rid of outdated comments from GHA for R CMD check

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
> `next` corresponds to the next version of R, either R-patched, or R-alpha, R-beta, R-rc, or R-prerelease

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
           # use 4.1 to check with rtools40's older compiler
           - {os: windows-latest, r: '4.1'}
 
-          # Use older ubuntu to maximise backward compatibility
```
Comment (IndrajeetPatil) [inline .github/workflows/R-CMD-check.yaml:37]: This is an outdated comment. We use `ubuntu-latest`.

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -72,7 +73,6 @@ jobs:
           http-user-agent: ${{ matrix.config.http-user-agent }}
           use-public-rspm: true
 
-      # `{patrick}` imports `{purrr}`, and so needs to be ignored on R < 3.5
```
Comment (IndrajeetPatil) [inline .github/workflows/R-CMD-check.yaml:75]: Not relevant anymore.

**Comment by MichaelChirico** [comment]
Is there any particular reason we need to run on `next` version for every change?

We already have "run on every R version" as part of the release procedure, is that not sufficient?

(changes to obsolete comments look good anyway)

**Comment by IndrajeetPatil** [comment]
> We already have "run on every R version" as part of the release procedure, is that not sufficient?

Yeah, that's fair. It does sound like an overkill to run it for every commit.

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**